### DISCUSSION TITLE
Error when trying to migrate project

### DISCUSSION THREAD

**Discussion by buidles** [supabase/supabase#20987] [Questions]
I have been developing locally and now is the time to deploy to a project on supabase.com.

I'm following the instruction [here](https://supabase.com/docs/guides/platform/migrating-and-upgrading-projects#migrate-your-project).

When running `supabase db dump --db-url "$OLD_DB_URL" -f roles.sql --role-only` I get the error 

```
Dumping roles from remote database...
pg_dumpall: error: connection to server on socket "/private/tmp/.s.PGSQL.5432" f
```

What could the issue be?

Edit:

It looks like the env variable are not being picked up e.g. when running `echo $OLD_DB_URL` from my local supabase project the value is not printed.

From the documentation: 

"Set environment variables for the old project's database URL as $OLD_DB_URL and the new project's as $NEW_DB_URL."

This instructions are frustratingly incomplete - where are these environment variables supposed to be set? I have tried adding an `.env` file to `supabase/functions`, `supabase`, and the project root but the variables are still not picked up.

Edit:

So in conclusion I have a local development instance of a supabase project that I would like to now deploy to supabase.com - what is the easiest way of doing this? Would need to include triggers, functions, etc. also.
